### ISSUE TITLE
Production Quality

### ISSUE THREAD

**Issue by dbettin** [KodrAus/nuget-rs#28]
Hey! Thanks for writing this crate!   

Our company would like to start writing some of our data heavy processing in rust and we would like to be able to call it from .net core. 

How do you feel about the level of stability/quality of the crate?  

Are there any examples of passing complex types (structs) and returning complex types?

**Comment by KodrAus** [comment]
Hi @dbettin thanks for reaching out! This crate is still very immature, right now it's just supporting some toy experiments of mine. I do have some very basic cross-platform support that needs to be published on `crates.io`, but there's a lot more work to be done adding support for finer grained .NET RIDs (right now we just support the top level `win`, `osx` and `linux`) and ensuring Rust dependencies are available.

But if all you need to do is bundle up Rust libraries for a single target then it might have enough in there for you already. If you do take a look and have any thoughts on improvements or things you'd expect then I'd love to hear them! I will continue working on this crate.

I haven't got any examples using complex structures myself, but I'd started playing with [sharing `Vec` buffers with C#](https://github.com/KodrAus/csharp-rust/tree/master/src/byte_array), which is using this crate. The new C# `Span` is great for this sort of thing. There's also an example of sharing datastructures in the [Rust FFI Omnibus](http://jakegoulding.com/rust-ffi-omnibus/objects/) that includes C#.

**Comment by dbettin** [comment]
Yes, we are only interested in a single target.  Additionally, I believe sending an array of bytes or Vec should be sufficient for our use case.

I will give it a try. Thanks again!
